### PR TITLE
add &= for seqs to system.nim

### PR DESCRIPTION
**Pull request by zielmicha**
As promised in http://forum.nim-lang.org/t/1762

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -3289,6 +3289,11 @@ proc `/=`*[T: float|float32](x: var T, y: T) {.inline, noSideEffect.} =
 
 proc `&=`* (x: var string, y: string) {.magic: "AppendStrStr", noSideEffect.}
 
+proc `&=`*[T](a: var seq[T], b: seq[T]) =
```
Comment (yglukhov) [inline lib/system.nim]: 1. `noSideEffect`?
2. extend `a` in advance?
3. `b` - openarray?

**Comment by dom96** [comment]
Is ``.add`` really so bad?

**Comment by Araq** [comment]
@dom96 it's about consistency. &= already exists for strings, it should work for seqs. And in fact, it should also work for files and streams.

File: lib/system.nim (hunk 1 above)
Comment (Araq) [inline lib/system.nim]: Er, no,  the signature needs to be: 

proc `&=`*[T](a: var seq[T]; b: T) = ...

**Comment by dom96** [comment]
@Araq or ``&=`` should be removed altogether :P

**Comment by Araq** [comment]
@dom96 That would be possible but Nim really needs a generic 'sink' operator so that I can switch between strings and files and streams without having to rewrite my code.

**Comment by dom96** [comment]
@Araq: Fair enough.

File: lib/system.nim (hunk 1 above)
Comment (dom96) [inline lib/system.nim]: Could ``b`` be an openarray?

**Comment by Araq** [comment]
Thanks, but I did it my way.

**Pull request closed without merging**